src/core/Toast/Toast.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { Component, forwardRef, ReactNode } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { baseStyles } from './Toast.baseStyles';
+import { SuomifiThemeProp, SuomifiThemeConsumer } from '../theme';
+import { Icon } from '../Icon/Icon';
+import { Heading } from '../Heading/Heading';
+import { HtmlDiv, HtmlDivWithRef } from '../../reset';
+import { hLevels } from '../../reset/HtmlH/HtmlH';
+
+export interface ToastProps {
+  /** Set aria-live mode for the Toast text content and label.
+   * @default 'polite'
+   */
+  ariaLiveMode?: 'polite' | 'assertive' | 'off';
+  /** Label heading for the Toast */
+  headingText?: string;
+  /** Main content of the Toast */
+  children?: ReactNode;
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Elements to be rendered under the Toast text. Can be e.g. buttons, links etc. */
+  headingVariant?: Exclude<hLevels, 'h1'>;
+  /** Id for focus */
+  id?: string;
+}
+interface InnerRef {
+  forwardedRef: React.RefObject<HTMLDivElement>;
+}
+const baseClassName = 'fi-toast';
+export const toastClassNames = {
+  styleWrapper: `${baseClassName}-wrapper`,
+  heading: `${baseClassName}-heading`,
+  contentWrapper: `${baseClassName}-content-wrapper`,
+  icon: `${baseClassName}_icon`,
+  iconWrapper: `${baseClassName}_icon-wrapper`,
+};
+class BaseToast extends Component<ToastProps & InnerRef> {
+  render() {
+    const {
+      ariaLiveMode = 'polite',
+      children,
+      className,
+      forwardedRef,
+      headingText,
+      headingVariant = 'h2',
+      id,
+      ...passProps
+    } = this.props;
+    return (
+      <HtmlDivWithRef
+        className={classnames(baseClassName, className)}
+        as="section"
+        {...passProps}
+      >
+        <HtmlDiv className={toastClassNames.styleWrapper}>
+          <HtmlDiv className={toastClassNames.iconWrapper}>
+            <Icon icon="checkCircle" className={toastClassNames.icon} />
+          </HtmlDiv>
+
+          <HtmlDiv
+            className={toastClassNames.contentWrapper}
+            id={id}
```

Comment:
The id should be either handled using AutoId or conditionally added. At the moment it can be empty as the prop is not mandatory.

Reply:
And then remember to update the JSDoc of the `id` prop when it is using the **AutoId**.

Reply:
Do we need id? There is no button where we should move focus...

Reply:
AutoId is not probably necessary here, but the whole prop should be conditional as there should be no empty id's. So if there is no id provided in props, there should be no id in DOM.

Reply:
There are no ids in dom if not given.

Reply:
Seems to work, but is this defined somewhere with React or JSX that this is actually how it should work rather than adding an empty attribute to DOM?

Reply:
Looking this more thoroughly revealed a potential bug. The id is now referenced in the close buttons aria-described by so that when a screen reader user navigates to the button, also the contents of the text are read to help understand what the close button is about. If no id is provided for the component, only close button text is read. This component need AutoId in my opinion.